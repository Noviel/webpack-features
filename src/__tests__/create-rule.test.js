import createRule from '../create-rule';

describe('createRule', () => {
  it('should throw with undefined loaders', () => {
    expect(() => createRule())
      .toThrow();
  });

  it('should throw with a non-object loaders', () => {
    expect(() => createRule('loaders'))
      .toThrow();
  });

  it('should assign default option values', () => {
    expect(createRule({ loader: 'loader' }))
      .toMatchSnapshot();
  });

  it('should merge provided options', () => {
    expect(
      createRule(
        { 
          loader: 'simple' 
        }, 
        {
          test: /\.jsx?$/
        }
      )
    )
      .toMatchSnapshot();
  });
});
