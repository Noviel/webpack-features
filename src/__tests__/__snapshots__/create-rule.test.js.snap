// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRule should assign default option values 1`] = `
Object {
  "exclude": /node_modules/,
  "use": Object {
    "loader": "loader",
  },
}
`;

exports[`createRule should merge provided options 1`] = `
Object {
  "exclude": /node_modules/,
  "test": /\\\\\\.jsx\\?\\$/,
  "use": Object {
    "loader": "simple",
  },
}
`;
